_posts/2020-08-24-training-imagenet-with-r/training-imagenet-with-r.Rmd

```diff
@@ -0,0 +1,270 @@
+---
+title: "Training ImageNet with R"
+description: >
+  This post explores how to train large datasets with TensorFlow and R. Specifically, we present how to download and repartition ImageNet, followed by training ImageNet across multiple GPUs in distributed environments using TensorFlow and Apache Spark.
+author:
+  - name: Javier Luraschi
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+date: 08-24-2020
+categories:
+  - R
+  - TensorFlow/Keras
+  - Distributed Computing
+  - Data Management
+bibliography: bibliography.bib
+preview: images/fishing-net.jpg
+output:
+  distill::distill_article:
+    self_contained: false
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+[ImageNet](http://www.image-net.org/) [@deng2009imagenet] is an image database organized according to the [WordNet](http://wordnet.princeton.edu/) [@miller1995wordnet] hierarchy which, historically, has been used in computer vision benchmarks and research. However, it was not until AlexNet [@krizhevsky2012imagenet] introduced deep learning concepts like convolutional-neural-networks and GPUs, that the computer vision discipline turned to deep learning to achieve state-of-the-art models that revolutionized their field. Given the importance of ImageNet and AlexNet, this post introduces the tools and techniques that you can consider using when training ImageNet and other large-scale datasets with R.
+
+Now, in order to process the ImageNet dataset, we will first have to -- divide and conquer -- by partitioning into several manageable subsets. Afterwards, we will train ImageNet using AlexNet across multiple GPUs and compute instances. [Preprocessing ImageNet](#preprocessing-imagenet) and [distributed training](#distributed-training) are the two topics that this post will present and discuss, starting with preprocessing ImageNet.
+
+## Preprocessing ImageNet
+
+When dealing with large datasets, even simple tasks like downloading or reading a dataset can be much harder than what you would expect. For instance, since ImageNet is roughly 300GB in size, you will need to make sure we have at least 600GB of free space to leave some room for download and decompression. But no worries, you can always borrow computers with huge disk drives from your favorite cloud provider. While you are at it, you should also request compute instances with multiple GPUs, Solid State Drives (SSDs), and a reasonable amount of CPUs and memory. If you want to use the exact configuration we used, take a look at the [mlverse/imagenet](https://github.com/mlverse/imagenet) repo, which contains a Docker image and configuration commands required to provision reasonable computing resources for this task. In summary, make sure you have access to sufficient compute resources.
```

Comment:
```suggestion
When dealing with large datasets, even simple tasks like downloading or reading a dataset can be much harder than what you would expect. For instance, since ImageNet is roughly 300GB in size, you will need to make sure to have at least 600GB of free space to leave some room for download and decompression. But no worries, you can always borrow computers with huge disk drives from your favorite cloud provider. While you are at it, you should also request compute instances with multiple GPUs, Solid State Drives (SSDs), and a reasonable amount of CPUs and memory. If you want to use the exact configuration we used, take a look at the [mlverse/imagenet](https://github.com/mlverse/imagenet) repo, which contains a Docker image and configuration commands required to provision reasonable computing resources for this task. In summary, make sure you have access to sufficient compute resources.
```